gpu-miner.c

```diff
@@ -148,91 +131,86 @@ int main() {
 
 	// Set OpenCL kernel arguments
 	ret = clSetKernelArg(kernel, 0, sizeof(cl_mem), (void *)&blockHeadermobj);
-	if (ret != CL_SUCCESS) { printf("failed to set first kernel arg: \n"); return -1; }
+	if (ret != CL_SUCCESS) { printf("failed to set first kernel arg: \n"); exit(1); }
 	ret = clSetKernelArg(kernel, 1, sizeof(cl_mem), (void *)&headerHashmobj);
-	if (ret != CL_SUCCESS) { printf("failed to set fifth kernel arg: \n"); return -1; }
+	if (ret != CL_SUCCESS) { printf("failed to set fifth kernel arg: \n"); exit(1); }
 	ret = clSetKernelArg(kernel, 2, sizeof(cl_mem), (void *)&targmobj);
-	if (ret != CL_SUCCESS) { printf("failed to set third kernel arg: \n"); return -1; }
+	if (ret != CL_SUCCESS) { printf("failed to set third kernel arg: \n"); exit(1); }
 	ret = clSetKernelArg(kernel, 3, sizeof(cl_mem), (void *)&nonceOutmobj);
-	if (ret != CL_SUCCESS) { printf("failed to set second kernel arg: \n"); return -1; }
+	if (ret != CL_SUCCESS) { printf("failed to set second kernel arg: \n"); exit(1); }
 	ret = clSetKernelArg(kernel, 4, sizeof(cl_mem), (void *)&nonceOutLockmobj);
-	if (ret != CL_SUCCESS) { printf("failed to set fourth kernel arg: \n"); return -1; }
+	if (ret != CL_SUCCESS) { printf("failed to set fourth kernel arg: \n"); exit(1); }
+	ret = clSetKernelArg(kernel, 5, sizeof(cl_mem), (void *)&iter_per_threadmobj);
+	if (ret != CL_SUCCESS) { printf("failed to set fourth kernel arg: \n"); exit(1); }
+
+	// Rough scan for 'optimal' thread count
+	double hash_rate, prev_hash_rate = 0;
+	global_item_size = 16;
+	iter_per_thread = 16 * 256;
+	while(1) {
+		global_item_size *= 2;
```

Comment:
you just double every time? Is a more fine-grained search better?

I'd also be curious if iter_per_thread made a difference.